orchestrator/migrations/helpers.py

```diff
@@ -446,6 +522,194 @@ def create(conn: sa.engine.Connection, new: Dict) -> None:
+                VALUES (:parent_id, :child_id)
+            """
+        ),
+        parent_id=parent_id,
+        child_id=child_id,
+    )
+
+
+def remove_product_block_relation_between_products_by_id(
+    conn: sa.engine.Connection, parent_id: str, child_id: str
+) -> None:
+    """Remove product block relation by id.
+
+    Args:
+        conn: DB connection as available in migration main file.
+        parent_id: ID of the parent product block.
+        child_id: ID of the child product block.
+
+    Usage:
+    ```python
+    parent_id = "parent_id"
+    child_id = "child_id"
+    remove_product_block_relation_between_products_by_id(conn, parent_id, child_id)
+    ```
+    """
+
+    conn.execute(
+        sa.text(
+            """
+                DELETE FROM product_block_relations
+                WHERE parent_id=:parent_id AND child_id=:child_id
+            """
+        ),
+        parent_id=parent_id,
+        child_id=child_id,
+    )
+
+
+def delete_resource_type_by_id(conn: sa.engine.Connection, id: str) -> None:
+    """Delete resource type by resource type id.
+
+    Args:
+        conn: DB connection as available in migration main file.
+        id: ID of the resource type to delete.
+
+    Usage:
+    ```python
+    resource_type_id = "id"
+    delete_resource_type_by_id(conn, resource_type_id)
+    ```
+    """
+    conn.execute(sa.text("DELETE FROM resource_types WHERE resource_type_id=:id"), id=id)
+
+
+def delete_resource_types_from_product_blocks(conn: sa.engine.Connection, delete: Dict) -> None:
+    """Delete resource type from product blocks.
+
+    Note: the resource_type itself will not be deleted.
+
+    Args:
+        conn: DB connection as available in migration main file
+        delete: dict of product_blocks and resource_types names that you want to unlink
+
+    Example:
+        >>> obsolete_stuff = {
+            "ProductBlockName1": {
+                "resource_type1": "Resource description"
+            },
+            "ProductBlockName2": {
+                "resource_type1": "Resource description",
+                "resource_type1": "Resource description"
+            }
+        }
+        >>> delete_resource_types_from_product_blocks(conn, obsolete_stuff)
+    """
+    for product_block_name, resource_types in delete.items():
+        conn.execute(
+            sa.text(
+                """DELETE FROM product_block_resource_types
+                   USING resource_types
+                   WHERE product_block_id = (SELECT product_block_id FROM product_blocks WHERE name=:product_block_name) AND
+                     resource_types.resource_type_id = product_block_resource_types.resource_type_id AND
+                     resource_types.resource_type IN :obsolete_resource_types"""
+            ),
+            product_block_name=product_block_name,
+            obsolete_resource_types=tuple(resource_types.keys()),
+        )
+
+
 def delete_resource_types(conn: sa.engine.Connection, delete: Dict) -> None:
```

Comment:
In our helpers the `delete` param has a different type;

```python
def delete_resource_types(conn: sa.engine.Connection, delete: Sequence[str]) -> None:
```

Here it's a Dict. It's changed to a sequence-like object further down by `tuple(delete)`, keeping the keys, but we might get mypy errors for this

Reply:
I will re-inspect this change: FWIW it passes type checking.